### PR TITLE
Dock icon magnification settings

### PR DESCRIPTION
**Pull request by pik694**
- add dock icon magnification settings

### CODE REVIEW

File: modules/system/defaults/dock.nix [hunk 1]
```diff
@@ -163,6 +163,23 @@ in {
...
+    system.defaults.dock.magnification = mkOption {
+      type = types.nullOr types.bool;
+      default = false;
+      description = lib.mdDoc ''
+        Magnify icon on hover.  The default is false.
+      '';
+    };
+
+    system.defaults.dock.largesize = mkOption {
+      type = types.nullOr types.int;
```
[the remark this answers is not in the document]
Comment (pik694) [reply modules/system/defaults/dock.nix:175]: for me the values can range from 16 up to 128 - I can add it in the documentation

[the remark this answers is not in the document]
Comment (Enzime) [reply modules/system/defaults/dock.nix:175]:
```suggestion
      type = types.nullOr (types.ints.between 16 128);
```

File: modules/system/defaults/dock.nix [hunk 2]
```diff
@@ -163,6 +163,24 @@ in {
...
+      description = lib.mdDoc ''
+        Magnify icon on hover. The default is false.
+      '';
+    };
+
+    system.defaults.dock.largesize = mkOption {
+      type = types.nullOr types.int;
+      default = null;
+      description = lib.mdDoc ''
+        Hover maginified icon size. The default is null.
```
Comment (Enzime) [inline modules/system/defaults/dock.nix:178]:
```suggestion
        Magnified icon size on hover.
```

File: modules/system/defaults/dock.nix [hunk 3]
```diff
@@ -163,6 +163,24 @@ in {
       '';
     };
 
+    system.defaults.dock.magnification = mkOption {
+      type = types.nullOr types.bool;
+      default = false;
```
Comment (Enzime) [inline modules/system/defaults/dock.nix:168]: Should we make `null` the default so we don't override this setting for users who don't set a value in `nix-darwin`?

**Review by Enzime — CHANGES_REQUESTED** [review]
Thanks for the PR :+1:

File: modules/system/defaults/dock.nix (hunk 3 above)
Comment (pik694) [reply modules/system/defaults/dock.nix:168]: sure, you're right! I have prepared these changes sometime ago, and should have pruned the code before creating PR 🙂

**Review by Enzime — APPROVED** [review]
LGTM

**Pull request merged**